src/app/features/tasks/recurring-interval/time-interval.ts

```diff
@@ -0,0 +1,37 @@
+import { unitOfTime } from "moment";
+
+/**
+ * A simple time interval representation for repetitions or offsets.
+ */
+export class TimeInterval {
+  static DATA_TYPE = "time-interval";
+
+  /** amount of the unit to be offset */
+  value: number;
+
+  /** unit (e.g. days, weeks) to be offset */
+  unit: unitOfTime.Base;
+}
+
+export function generateLabelFromInterval(interval: TimeInterval) {
+  return (
+    " " +
+    $localize`:custom interval select option:every ${interval.value} ` +
+    timeunitLabelMap.get(interval.unit)
+  );
```

Comment:
```suggestion
  return (
    " " +
    $localize`:custom interval select option| e.g. every 2 weeks:every ${interval.value} ${timeunitLabelMap.get(interval.unit)}`
  );
```
This way a re-ordering can also be achieved